src/main/java/cloud/fogbow/common/util/SystemUserUtil.java

```diff
@@ -1,25 +1,29 @@
 package cloud.fogbow.common.util;
 
-import cloud.fogbow.common.constants.FogbowConstants;
 import cloud.fogbow.common.constants.Messages;
 import cloud.fogbow.common.exceptions.UnexpectedException;
 import cloud.fogbow.common.models.SystemUser;
-import com.google.gson.reflect.TypeToken;
-import org.apache.commons.lang.StringUtils;
 
 public class SystemUserUtil {
-    public static String serialize(SystemUser systemUser) {
-        String className = systemUser.getClass().getName();
-        String serializedSystemUser = className + FogbowConstants.CLASS_NAME_SEPARATOR + GsonHolder.getInstance().toJson(systemUser);
+
+    public static final int SERIALIZED_SYSTEM_USER_MAX_SIZE = 2048;
+
+    public static String serialize(SystemUser systemUser) throws UnexpectedException{
+        SerializableEntity<SystemUser> serializableSystemUser = new SerializableEntity<SystemUser>(systemUser);
+        String serializedSystemUser = GsonHolder.getInstance().toJson(serializableSystemUser);
+
+        if(serializedSystemUser.length() > SystemUserUtil.SERIALIZED_SYSTEM_USER_MAX_SIZE) {
+            throw new UnexpectedException(Messages.Exception.MAXIMUM_USER_SIZE_EXCEEDED);
+        }
+
         return serializedSystemUser;
```

Comment:
If you decide to do what I suggested, you just need to do `new SerializableEntity<SystemUser>(systemUser).toString()` here.

Reply:
I've agreed with the first suggestion, but I need to keep a reference to the instance once besides return it I need to check its size...